src/system.transactions/tests/LTMEnlistmentTests.cs

```diff
@@ -0,0 +1,192 @@
+using Xunit;
+
+namespace System.Transactions.Tests
+{
+    public class LTMEnlistmentTests
+    {
+        public LTMEnlistmentTests()
+        {
+        }
+
+        [Theory]
+        [InlineData(1, EnlistmentOptions.None, Phase1Vote.Prepared, SinglePhaseVote.Committed, true, EnlistmentOutcome.Committed, TransactionStatus.Committed)]
+        [InlineData(2, EnlistmentOptions.None, Phase1Vote.Prepared, SinglePhaseVote.Committed, true, EnlistmentOutcome.Committed, TransactionStatus.Committed)]
+        [InlineData(1, EnlistmentOptions.None, Phase1Vote.Prepared, SinglePhaseVote.Aborted, true, EnlistmentOutcome.Aborted, TransactionStatus.Aborted)]
+        [InlineData(2, EnlistmentOptions.None, Phase1Vote.Prepared, SinglePhaseVote.Aborted, true, EnlistmentOutcome.Aborted, TransactionStatus.Aborted)]
+        [InlineData(1, EnlistmentOptions.EnlistDuringPrepareRequired, Phase1Vote.Prepared, SinglePhaseVote.Committed, true, EnlistmentOutcome.Committed, TransactionStatus.Committed)]
+        [InlineData(2, EnlistmentOptions.EnlistDuringPrepareRequired, Phase1Vote.Prepared, SinglePhaseVote.Committed, true, EnlistmentOutcome.Committed, TransactionStatus.Committed)]
+        public void SinglePhaseDurable(int volatileCount, EnlistmentOptions volatileEnlistmentOption, Phase1Vote volatilePhase1Vote, SinglePhaseVote singlePhaseVote, bool commit, EnlistmentOutcome expectedVolatileOutcome, TransactionStatus expectedTxStatus)
+        {
+            Transaction tx = null;
+            try
+            {
+                using (TransactionScope ts = new TransactionScope())
+                {
+                    tx = Transaction.Current.Clone();
+
+                    if (volatileCount > 0)
+                    {
+                        TestSinglePhaseEnlistment[] volatiles = new TestSinglePhaseEnlistment[volatileCount];
+                        for (int i = 0; i < volatileCount; i++)
+                        {
+                            // It doesn't matter what we specify for SinglePhaseVote.
+                            volatiles[i] = new TestSinglePhaseEnlistment(volatilePhase1Vote, SinglePhaseVote.InDoubt, expectedVolatileOutcome);
+                            tx.EnlistVolatile(volatiles[i], volatileEnlistmentOption);
+                        }
+                    }
+
+                    // Doesn't really matter what we specify for EnlistmentOutcome here. This is an SPC, so Phase2 won't happen for this enlistment.
+                    TestSinglePhaseEnlistment durable = new TestSinglePhaseEnlistment(Phase1Vote.Prepared, singlePhaseVote, EnlistmentOutcome.Committed);
+                    tx.EnlistDurable(Guid.NewGuid(), durable, EnlistmentOptions.None);
+
+                    if (commit)
+                    {
+                        ts.Complete();
+                    }
+                }
+            }
+            catch (TransactionInDoubtException)
+            { }
+            catch (TransactionAbortedException)
+            { }
```

Comment:
Same question as earlier as to whether we should be more specific about what exceptions we're expecting and when... I'll stop asking :)